include/tvm/relay/op_attr_types.h

```diff
@@ -81,10 +81,16 @@ using TOpIsStateful = bool;
  */
 using TNonComputational = bool;
 
+enum ShapeDependantKind {
+  kShapeDependantShape = 0,
+  kShapeDependantData = 1,
+  kShapeDependantBoth = 2,
```

Comment:
A bit curious about whether we need kShapeDependantBoth. Currently we can support data dependent shape function, and reshape with symbolic newshape should also be data dependent.

Reply:
Also this change involves a lot of changes in compile engine as well. AFAIK, dynamic stridedslice doesn't require these changes. Dynamic reshape should not be that different?

Reply:
I checked #4312, and tried to use data.shape the same ways as _strided_slice_shape_func(), but I got this error from hybrid script parser
```
ValueError: All indices are supposed to be constants
```
it seems _strided_slice_shape_func only uses const index when accessing data.shape, but _reshape_shape_func uses variables as indices to access data_shape

I'd appreciate it if you can give me some hint about how to solve this.

Reply:
https://github.com/apache/incubator-tvm/pull/4312/files#diff-81885bc0d9c067a058536e5a22ff6b23R108 strided_slice_shape_func iterates data shape over ndim. I think reshape should just do quite similar thing.

Reply:
https://github.com/apache/incubator-tvm/blob/6e1cd8256e461deca51e45e7aae33088e8f5b966/python/tvm/relay/op/_transform.py#L179
This line uses a range() and I think it causes variable index.

I tried to allocate a data shape tensor by tvm.te.placeholder(), but it can't be passed to tvm.lower() when this shape func is lowered.
But data shape tensor allocated in this line will be stored in CachedNode object (https://github.com/apache/incubator-tvm/blob/2c1ca60ea38587401a20f11c5e64f452b79fa777/src/relay/backend/compile_engine.cc#L348)
Please correct me if I missunderstood your suggestion about allocating a buffer.

Reply:
You can allocate buffer inside hybrid script.

Reply:
Thanks, it's solved now